### ISSUE TITLE
Sunfish can’t handle end games, like KRK.

### ISSUE THREAD

**Issue by ChristerNilsson** [thomasahle/sunfish#116]
I was impressed by the strong opening and middle game, but not so much by the end game. I guess PST for endgames is needed and some interpolation. KRK is considered easy, but I failed solving this by myself. Maybe mobility could also be helpful.

**Comment by thomasahle** [comment]
This is true. I wonder if Micro-Max (https://www.chessprogramming.org/Micro-Max) can solve KRK?

**Comment by thomasahle** [comment]
Some options for the endgame weakness, for discussion:

1. **Tapered eval**: a second set of endgame PSTs, interpolated by a game-phase count (remaining material). This is the standard fix — the current tables literally don't know the king should centralize in the endgame. Costs a second 384-value table (though it compresses well) and a little speed.
2. **Cheap mop-up term** for basic mates (KRK/KQK): when the opponent has a bare king, add a bonus for driving it toward the edge/corner and for our king approaching. ~5 lines, targeted at exactly the reported failure, no effect on the rest of the game.
3. **Mobility**: helps broadly but is expensive to compute in Python — probably a poor trade at sunfish's node rates.
4. **Nothing but depth**: MTD-bi does find KRK mates when it gets deep enough; the failure is partly just node throughput. Any speed improvement (pypy, the mutable-board work in #39) shrinks this problem for free.

Re Micro-Max: yes, it solves KRK — it has explicit king-centralization pressure in its eval, which is essentially option (2). My inclination: (2) now if we want a quick win, (1) as the principled fix. No action taken yet, per discussion.

**Comment by thomasahle** [comment]
We now have a special end-game table for kings. We also have unit tests for KRK and KQK.
